UefiTestingPkg/AuditTests/PagingAudit/Windows/MemoryRangeObjects.py

```diff
@@ -135,10 +147,46 @@ def GetSystemMemoryType(self):
         if self.SystemMemoryType is None:
             return "None"
         try:
+            if (self.SystemMemoryType == 5 or self.SystemMemoryType == 6 or self.SystemMemoryType == 7):
+                return MemoryRange.SystemMemoryTypes[self.SystemMemoryType].format(self.CpuNumber)
+
             return MemoryRange.SystemMemoryTypes[self.SystemMemoryType]
         except:
             raise Exception("System Memory Type is invalid %d" % self.SystemMemoryType)
 
+    def StackInit (self, SystemMemoryType, PhysicalStart, Length):
+        self.PhysicalStart = PhysicalStart
+        self.PhysicalSize = Length
+
+        if SystemMemoryType == "StackGuard":
+            self.SystemMemoryType = 3
+        elif SystemMemoryType == "Stack":
+            self.SystemMemoryType = 4
+        else:
+            raise Exception("System Memory Type is invalid %s" % SystemMemoryType)
+
+    def ApStackInit (self, SystemMemoryType, PhysicalStart, Length, CpuNumber):
+        self.PhysicalStart = PhysicalStart
+        self.PhysicalSize = Length
+        self.CpuNumber = CpuNumber
+        
+        if SystemMemoryType == "ApStackGuard":
+            self.SystemMemoryType = 5
+        elif SystemMemoryType == "ApStack":
+            self.SystemMemoryType = 6
+        elif SystemMemoryType == "ApSwitchStack":
+            self.SystemMemoryType = 7
+        else:
+            raise Exception("System Memory Type is invalid %s" % SystemMemoryType)
+    
+    def NullInit (self, SystemMemoryType, PhysicalStart):
+        self.PhysicalStart = PhysicalStart
+        self.PhysicalSize = MemoryRange.PageSize["4k"]

```

Comment:
why is this hard coded to 4k?  Why isn't that data in the info from the SUT

Reply:
What is the SUT?

I set it to 4K here because the page granularity is spec defined as 4K in UEFI. If the NULL page hasn't been fully split to 4K granularity, then the report will show the page size for the NULL page as 2m(p) per the logic you added a while back: https://github.com/microsoft/mu_plus/commit/c6f237d4281a2e3d3700a5853a5eb2f120116d33